### PR TITLE
Add metrics labels to all metrics

### PR DESCRIPTION
**Pull request by huikang**
Cherry pick from the [metrics-label-backport branch](https://github.com/hashicorp/memberlist/commits/metrics-labels-backport)

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -244,10 +244,14 @@ type Config struct {
...
 	// a message to that node.
 	RequireNodeNames bool
+
 	// CIDRsAllowed If nil, allow any connection (default), otherwise specify all networks
 	// allowed to connect (you must specify IPv6/IPv4 separately)
 	// Using [] will block all connections.
 	CIDRsAllowed []net.IPNet
+
+	// MetricLabels is a map of optional labels to apply to all metrics emitted.
+	MetricLabels map[string]string
```
Comment (rboyer) [inline config.go:254]: Would it make sense to match the serf config and have this be a slice of metric labels?

Comment (huikang) [reply config.go:254]: Either way looks good to me. If we switch to `MetricLabels []metrics.Label`, we can get rid of `mapToLabels` and pass []metrics.Label all the way from consul to memberlist, right?

Comment (rboyer) [reply config.go]: yep

File: memberlist.go [hunk 2]
```diff
@@ -198,6 +203,8 @@ func newMemberlist(conf *Config) (*Memberlist, error) {
 		}
 	}
 
+	metricLabels := conf.MetricLabels
```
Comment (rboyer) [inline memberlist.go:206]: nit: why make a local variable for two uses below?

**Review by rboyer — APPROVED** [review]

File: memberlist.go (hunk 2 above)
Comment (huikang) [reply memberlist.go]: oops, I forgot to delete the variables after change the type; we used to have `metricLabels := mapToLabels(conf.MetricLabels)`

Fixed now.

**Pull request merged**